### DISCUSSION TITLE
PortableGit credential.helper=helper-selector override

### DISCUSSION THREAD

**Discussion by viceice** [git-for-windows/git#3410] [Q&A] [answered]
Hi I'm using the `PortableGit` version installed via `scoop`. Every time i update git, the global helper is reset to `helper-selector` at `<git-dir>/etc/gitconfig`, which is expected.

So i've added this to my `~/.gitconfig` and expect that this will override the global config:
```ini
[credential]
  helper = manager-core
```

But it won't, so is this expected with the portable version?

So my workaround for now is to run `git config --system credential.helper manager-core` after git update. 😕

**Comment by rimrul** [comment, marked as answer]
> [credential] 	
> helper = helper-selector

I assume that's a typo and you meant `manager-core`.

You can indeed override the system setting, but it's slightly more complicated.

From [gitcredentials(7)](http://git-scm.com/docs/gitcredentials#_configuration_options):

> If there are multiple instances of the credential.helper configuration variable, each helper will be tried in turn, and may provide a username, password, or nothing. Once Git has acquired both a username and a password, no more helpers will be tried.

> If credential.helper is configured to the empty string, this resets the helper list to empty (so you may override a helper set by a lower-priority config file by configuring the empty-string helper, followed by whatever set of helpers you would like).

So this should work:

```ini
[credential] 	
    helper = ""
    helper = manager-core
```

**Reply by viceice** [reply]
yes, there was a type, fixed it. this seems to work too

```ini
[credential] 	
    helper =
    helper = manager-core
```

**Reply by viceice** [reply]
Note: From powershell you need to call 
```pwsh
git config --global --replace-all credential.helper '""'
git config --global --add credential.helper manager-core
```
